spec/try_spec.rb

```diff
@@ -102,13 +103,20 @@
     let(:goal) { { kind: 'last_query_passes' } }
 
     context 'and query that passes' do
-      let(:request) { struct query: '123', goal: goal }
+      let(:query) { '123' }
       it { expect(result[1]).to eq :passed }
     end
 
     context 'and query that fails' do
-      let(:request) { struct query: 'asdasd', goal: goal }
+      let(:query) { 'asdasd' }
       it { expect(result[1]).to eq :failed }
     end
   end
-end
\ No newline at end of file
+
+  context 'when query errors result is not blank' do
+    let(:goal) { { kind: 'last_query_fails' } }
+    let(:query) { 'foo()' }
+
+    it { expect(result[2][:result]).to include 'ReferenceError: foo is not defined' }
```

Comment:
Just to be sure, can we add a test that asserts that the status this `failed`?

Reply:
Sounds good!